courses/templates/course_list_to_add.html

```diff
@@ -12,12 +12,14 @@
     <th>Add</th>
   </tr>
   {% for c in courses %}
+  {% if c.student_registration_open %}
```

Comment:
in the templates .. use if statements to hide and show links .. not to do business logic

for example, link to join course, only show it for student
link to edit course or perform instructor actions, only show for instructor